### PR TITLE
feat(elements): Add telemetry

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Adds telemetry tracking to better understand what people build with Elements. Initially we'll be able to see:

- If a user creates a sign-in and/or sign-up flow
- Which steps they create
- Which inputs they use inside their steps

Fixes SDK-1658

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/elements/src/react/common/form/index.tsx [hunk 1]
```diff
@@ -528,6 +530,10 @@ type FormInputProps =
  */
 const Input = React.forwardRef<React.ElementRef<typeof RadixControl>, FormInputProps>(
   (props: FormInputProps, forwardedRef) => {
+    const telemetry = useTelemetry();
+
+    telemetry?.record(eventComponentMounted('Elements_Input', props));
+
```
Comment (panteliselef) [inline packages/elements/src/react/common/form/index.tsx:536]: ❓
1. Wouldn't this fire on every re-render ?
2. Seeing the implementation of `eventComponentMounted` the props you seem to be interested will be ignored.

Comment (LekoArts) [reply packages/elements/src/react/common/form/index.tsx]: 1. Yes. https://github.com/clerk/javascript/pull/3267 is fixing this
2. Thanks! I fixed it by introducing a new helper function

File: packages/shared/src/telemetry/events/component-mounted.ts [hunk 2]
```diff
@@ -11,8 +11,19 @@ type EventComponentMounted = {
...
 /**
  * Fired when one of the Clerk components is mounted.
+ *
+ * @param component - The name of the component.
+ * @param props - The props passed to the component. Will be filtered to a known list of props.
+ *
+ * @example
+ * telemetry.record(eventComponentMounted('SignUp', props));
  */
 export function eventComponentMounted(
```
Comment (BRKalow) [inline packages/shared/src/telemetry/events/component-mounted.ts:28]: can we change this one to `eventPrebuiltComponentMounted` (or something similar) and expose your currently named `eventComponentMountedRaw` as `eventComponentMounted`?

Comment (LekoArts) [reply packages/shared/src/telemetry/events/component-mounted.ts]: Done 👍

File: packages/elements/src/react/common/form/index.tsx (hunk 1 above)
Comment (LekoArts) [reply packages/elements/src/react/common/form/index.tsx]: They will probably be released around the same time so it's not blocking

**Review by BRKalow — APPROVED** [review]
⭐

**Review by tmilewski — APPROVED** [review]

**Pull request merged**